From 88d95e108e40dcf666d725f2c52c43f7dcd5321c Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 24 Aug 2021 14:57:17 +0200
Subject: [PATCH] Fix build with glibc-2.34

* fixes:
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.cpp:25:29: error: no declaration matches 'std::shared_ptr<anbox::common::MountEntry> anbox::common::MountEntry::create(const boost::filesystem::path&, const boost::filesystem::path&, const string&, long unsigned int, const string&)'
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.cpp:47:29: error: no declaration matches 'std::shared_ptr<anbox::common::MountEntry> anbox::common::MountEntry::create(const std::shared_ptr<anbox::common::LoopDevice>&, const boost::filesystem::path&, const string&, long unsigned int, const string&)'
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.cpp:57:29: error: no declaration matches 'std::shared_ptr<anbox::common::MountEntry> anbox::common::MountEntry::create(const boost::filesystem::path&)'
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.h:27:15: error: 'shared_ptr' in namespace 'std' does not name a template type
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.h:30:15: error: 'shared_ptr' in namespace 'std' does not name a template type
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.h:33:15: error: 'shared_ptr' in namespace 'std' does not name a template type
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/src/anbox/common/mount_entry.h:41:8: error: 'shared_ptr' in namespace 'std' does not name a template type

  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/external/xdg/xdg.cpp:179:44: error: no declaration matches 'std::shared_ptr<xdg::BaseDirSpecification> xdg::BaseDirSpecification::create()'
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/external/xdg/xdg.h:94:17: error: 'shared_ptr' in namespace 'std' does not name a template type
  anbox/3.0+gitAUTOINC+9de4e87cdd-r0/git/external/xdg/xdg_test.cpp:105:50: error: 'create' is not a member of 'xdg::BaseDirSpecification'
  ...

Upstream-Status: Submitted [https://github.com/anbox/anbox/pull/1892]

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 external/xdg/xdg.h             | 1 +
 src/anbox/common/mount_entry.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/external/xdg/xdg.h b/external/xdg/xdg.h
index 379f0ff..cb28271 100644
--- a/external/xdg/xdg.h
+++ b/external/xdg/xdg.h
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <vector>
+#include <memory>
 
 namespace xdg
 {
diff --git a/src/anbox/common/mount_entry.h b/src/anbox/common/mount_entry.h
index 647bae2..3a6cc9a 100644
--- a/src/anbox/common/mount_entry.h
+++ b/src/anbox/common/mount_entry.h
@@ -19,6 +19,7 @@
 #define ANBOX_COMMON_MOUNT_ENTRY_H_
 
 #include <boost/filesystem/path.hpp>
+#include <memory>
 
 namespace anbox::common {
 class LoopDevice;
